### PR TITLE
chore: add OpenSearch version 2.13

### PR DESCRIPTION
**Pull request by jwoehrle**
### Issue # (if applicable)

N/A

### Reason for this change

Add constants for newer OpenSearch versions (2.12 and 2.13)

### Description of changes

N/A

### Description of how you validated changes

No

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-opensearchservice/lib/version.ts [hunk 1]
```diff
@@ -99,6 +99,12 @@ export class EngineVersion {
   /** AWS OpenSearch 2.11 */
   public static readonly OPENSEARCH_2_11 = EngineVersion.openSearch('2.11');
 
+  /** AWS OpenSearch 2.12 */
+  public static readonly OPENSEARCH_2_12 = EngineVersion.openSearch('2.12');
```
Comment (TheRealAmazonKendra) [inline packages/aws-cdk-lib/aws-opensearchservice/lib/version.ts:103]: I don't see 2.12 included in the [supported versions](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/what-is.html#choosing-version).

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Can you please provide verification of support for 2.12?

File: packages/aws-cdk-lib/aws-opensearchservice/lib/version.ts (hunk 1 above)
Comment (jwoehrle) [reply packages/aws-cdk-lib/aws-opensearchservice/lib/version.ts]: my bad, I just assumed it was continuous.

Removed 2.12 now

**Comment by jwoehrle** [comment]
Already implemented in the meantime: https://github.com/aws/aws-cdk/pull/30399

**Pull request closed without merging**